php/Block.php

```diff
@@ -63,66 +63,213 @@ public function register_block() {
+			/**
+			 * If our current post (`get_the_ID()`) is returned in the query results,
+			 * remove it. This mimics the functionality of using the `post__not_in`
+			 * argument on the query.
+			 *
+			 * @see https://docs.wpvip.com/technical-references/code-quality-and-best-practices/using-post__not_in/
+			 */
+			if ( in_array( $current_post_id, $posts_list_query->posts, true ) ) {
+
+				// Remove the current post id by using `array_diff`, then reset the
+				// keys using `array_values` to ensure no gaps.
+				$posts_list_query->posts = array_values( array_diff( $posts_list_query->posts, [ $current_post_id ] ) );
+
+				// Update the post count and found post values now that the current
+				// post ID has been removed.
+				$posts_list_query->post_count    = $posts_list_query->post_count - 1;
+				$posts_list_query->found_posts   = $posts_list_query->found_posts - 1;
+			}
+
+			// Cache for 5 minutes.
+			set_transient( $cache_key, $posts_list_query->posts, MINUTE_IN_SECONDS * 5 );
+
+		} else {
+			/**
+			 * Use the cached post ids to construct a real WP_Query. This is a
+			 * bit more work, but for cache storage reasons, we don't want to
+			 * store the entire WP_Query object, just the relevant data.
+			 */
+			$posts_list_query = new WP_Query(
+				[
+					'post__in'       => $posts_list_query_post_ids,
+					'post_type'      => [ 'post', 'page' ], // Post types need to match the original query.
+					'fields'         => 'ids',
+				    'orderby'        => 'post__in', // Ensure proper order is maintained.
+				    'posts_per_page' => count( $posts_list_query_post_ids ), // Ensure all ids are used.
+				]
+			);
+		}
+
+		// Block render callbacks expect a string, so begin capturing any direct
+		// output.
+		ob_start();
+		?>
+		<div class="<?php echo esc_attr( $attributes['className'] ?? '' ); ?>">
+			<h2><?php esc_html_e( 'Post Counts', 'site-counts' ); ?></h2>
 
 			<?php
-			$query = new WP_Query(  array(
-				'post_type' => ['post', 'page'],
-				'post_status' => 'any',
-				'date_query' => array(
-					array(
-						'hour'      => 9,
-						'compare'   => '>=',
-					),
-					array(
-						'hour' => 17,
-						'compare'=> '<=',
-					),
-				),
-                'tag'  => 'foo',
-                'category_name'  => 'baz',
-				  'post__not_in' => [ get_the_ID() ],
-			));
-
-			if ( $query->have_posts() ) :
+			/**
+			 * Display a list of public post types and how many posts each has.
+			 */
+			if ( ! empty( $public_post_type_counts ) ) :
 				?>
-				 <h2>5 posts with the tag of foo and the category of baz</h2>
-                <ul>
-                <?php
-
-                 foreach ( array_slice( $query->posts, 0, 5 ) as $post ) :
-                    ?><li><?php echo $post->post_title ?></li><?php
-				endforeach;
+					<ul>
+						<?php foreach ( $public_post_type_counts as $values ) : ?>
+							<li>
+								<?php
+								echo esc_html(
+									sprintf(
+										__( 'There are %1$d %2$s.', 'site-counts' ),
```

Comment:
Again, this should use proper pluralization for `"There are [0,2,3,etc.]`" and `"There is [1]"`.